airflow/providers/ssh/operators/ssh.py

```diff
@@ -77,9 +87,24 @@ def __init__(
         self.remote_host = remote_host
         self.command = command
         self.timeout = timeout
+        self.conn_timeout = conn_timeout
+        self.cmd_timeout = cmd_timeout
+        if self.conn_timeout is None:
+            self.conn_timeout = self.timeout if self.timeout else TIMEOUT_DEFAULT
```

Comment:
I think by setting a default here, we effectively break allowing the connections `conn_timeout` to be used over the default. Maybe this would be better? Btw, I think we need better test coverage on default / extra handling like the hook too.

```suggestion
        if self.conn_timeout is None and self.timeout:
            self.conn_timeout = self.timeout
```

This'll let `None` be passed into the hook, which will let the connection extras be used and we avoid having the "default" set in 2 places.

Reply:
Good catch! A test around this case would have caught this, too, so my fault for not creating one. I'll tackle both of those tonight.